variable/reduction.cpp

```diff
@@ -38,13 +51,9 @@ void nansum_impl(const VariableView &summed, const VariableConstView &var) {
 
 template <typename Op>
 Variable sum_with_dim_impl(Op op, const VariableConstView &var, const Dim dim) {
-  auto dims = var.dims();
-  dims.erase(dim);
   // Bool DType is a bit special in that it cannot contain it's sum.
   // Instead the sum is stored in a int64_t Variable
-  Variable summed{is_dtype_bool(var) ? makeVariable<int64_t>(Dimensions(dims))
-                                     : copy(var.slice({dim, 0}))};
-  fill_zeros(summed);
+  auto summed = make_accumulant(var, dim, FillValue::ZeroNotBool);
```

Comment:
I'm unclear from the PR description where the original bug was found. Is this were there error was in the slice with   0-D data?

Reply:
Yes, `slice({dim, 0})` was an out-of-bounds access.